addons/calendar/models/calendar_recurrence.py

```diff
@@ -140,19 +140,21 @@ def _compute_name(self):
             else:
                 end = ''
 
-            if recurrence.rrule_type == 'weeky':
+            if recurrence.rrule_type == 'weekly':
                 weekdays = recurrence._get_week_days()
-                weekday_fields = (self._fields[weekday_to_field(w)] for w in weekdays)
-                on = _("on %s,") % ", ".join([field.string for field in weekday_fields])
+                # Convert Weekday object
+                weekdays = [str(w) for w in weekdays]
+                day_strings = [d[1] for d in WEEKDAY_SELECTION if d[0] in weekdays]
+                on = _("on %s") % ", ".join([day_name for day_name in day_strings])
```

Comment:
@arj-odoo Since I can change the logic in the legacy code, I think these codes can perfectly solve the problem